src/Compilers/CSharp/Test/Emit2/Semantics/PatternMatchingTests5.cs

```diff
@@ -3356,5 +3344,51 @@ bool M0({type} x0)
                 Diagnostic(ErrorCode.ERR_ConstantExpected, expression).WithLocation(6, 22)
             );
         }
+
+        [Fact]
+        public void IsPatternExpressionWithAwait()
+        {
+            var source = """
+using System;
+using System.Threading.Tasks;
+
+Console.WriteLine(await Async1(30));
+Console.WriteLine(await Async1(40));
+Console.WriteLine(await Async1(50));
+
+Console.WriteLine(await Async2(30));
+Console.WriteLine(await Async2(40));
+Console.WriteLine(await Async2(50));
+
+Console.WriteLine(await Async3(30));
+Console.WriteLine(await Async3(40));
+Console.WriteLine(await Async3(50));
+
+static async Task<bool> Async1(int i) {
+    // sub-expression
+    return await Task.FromResult(false) || i is 30 or 40;
+}
+static async Task<bool> Async2(int i) {
+    // input-expression
+    return await Task.FromResult(i) is 30 or 40;
```

Comment:
>return await Task.FromResult(i) is 30 or 40; [](http://example.com/codeflow?start=4&length=44)

I think it would be good to verify IL of MoveNext method for this scenario.